src/goto-programs/validate_goto_model.cpp

```diff
@@ -0,0 +1,251 @@
+/*******************************************************************\
+
+Module: Validate Goto Programs
+
+Author: Diffblue
+
+Date: Oct 2018
+
+\*******************************************************************/
+#include "validate_goto_model.h"
+
+#include <set>
+
+#include "goto_functions.h"
+#include <util/invariant.h>
+
+class validate_goto_modelt
+{
+public:
+  using function_mapt = goto_functionst::function_mapt;
+
+  validate_goto_modelt(
+    const goto_functionst &goto_functions,
```

Comment:
Why does this take `goto_functionst` when it says `validate_goto_model`?

Reply:
I passed in the highest level structure that is really needed. 
(intended as a form of documentation)

It is called goto_model as the command-line parameter is " --validate-goto-model"
I felt if I did things otherwise - that would _also_ be confusing. 

Open to suggestions.

Reply:
Ok, but I suspect you will eventually need to take a full `goto_modelt` to do some namespace lookups. But it's ok to leave it as is for now if you don't need that at the moment.